runtime/doc/lua-guide.txt

```diff
@@ -0,0 +1,760 @@
+*lua-guide.txt*                        Nvim
+
+                            NVIM REFERENCE MANUAL
+
+                          Guide to using Lua in Nvim
+
+
+                                       Type |gO| to see the table of contents.
+
+==============================================================================
+Introduction                                                         *lua-guide*
+
+This guide will go through the basics of using Lua in Neovim. It is not meant
+to be a comprehensive encyclopedia of all available features, nor will it
+detail all intricacies. Think of it as a survival kit - the bare minimum
+needed to know to comfortably get started on using Lua in Neovim.
+
+An important thing to note is that this isn't a guide to the Lua language
+itself. Rather, this is a guide on how to configure and modify Neovim through
+the Lua language and the functions we provide to help with this. Take a look
+at |luaref| and |lua-concepts| if you'd like to learn more about Lua itself.
+Similarly, this guide assumes some familiarity with the basics of Neovim
+(commands, options, mappings, autocommands), which are covered in the
+|user-manual|.
+
+------------------------------------------------------------------------------
+Some words on the API                                            *lua-guide-api*
+
+The purpose of this guide is to introduce the different ways of interacting
+with Neovim through Lua (the "API"). This API consists of three different
+layers:
+
+1. The "Vim API" inherited from Vim: |ex-commands| and |builtin| as well as
+|user-function|s in Vimscript. These are accessed through |vim.cmd()| and
+|vim.fn| respectively, which are discussed under |lua-guide-vimscript| below.
+
+2. The "Neovim API" written in C for use in remote plugins and GUIs; see |api|.
```

Comment:
Is the mention of C needed here?

Reply:
Maybe not needed, but deliberate: It _is_ one of the main differentiators to the Lua API (which is written in Lua), and carries the connotation "fast" (in contrast to the "Vim API" above).

Reply:
(That whole section is still under discussion anyway; it's just a suggestion so far.)